src/main/java/org/opensearch/ad/transport/AnomalyDetectorJobTransportAction.java

```diff
@@ -102,7 +102,7 @@ protected void doExecute(Task task, AnomalyDetectorJobRequest request, ActionLis
                 detectorId,
                 filterByEnabled,
                 listener,
-                () -> executeDetector(listener, detectorId, seqNo, primaryTerm, rawPath, requestTimeout, user),
+                (anomalyDetector) -> executeDetector(listener, detectorId, seqNo, primaryTerm, rawPath, requestTimeout, user),
```

Comment:
Curiously, it seems the anomalyDetector was not used in the executeDetector method?

Reply:
Yes, actually the `executeDetector` will get detector with detectorId. This PR is mainly to fix the security issue. Will refactor code in separate PR to reuse detector rather than query it again.

Reply:
Thanks!

Reply:
Create an issue to track this https://github.com/opensearch-project/anomaly-detection/issues/127